Faraday_Proxy_Tools/FaradayIO/deviceconfig.py

```diff
@@ -163,13 +175,70 @@ def update_gps(self, gps_boot_bitmask, latitude_str, latitude_dir_str, longitude
 
         :return: Nothing
         """
+        # Setup check variables
+        lat_check = True
+        lat_dir_check = True
+        lon_check = True
+        lon_dir_check = True
+        alt_check = True
+        alt_units_check = True
+
+        # Check Latitude formatting
+
         lat_check = len(latitude_str) <= self.MAX_GPS_LATITUDE_LEN
         lat_dir_check = len(latitude_dir_str) <= self.MAX_GPS_LATITUDE_DIR_LEN
+        latitude_str_format = latitude_str.split('.')
+
+        # Check for non numeric inputs
+        for item in latitude_str_format:
+            if not item.isdigit():
+                lat_check = False # non-numeric input - Fail
+
+        if len(latitude_str_format) != 2 or len(latitude_str_format[0]) > self.MAX_GPS_LATITUDE_LEADING_LEN or len(latitude_str_format[1]) > self.MAX_GPS_LATITUDE_TRAILING_LEN:
+            lat_check = False
+        else:
+            #Correct latitude formatting
+            latitude_str_format = latitude_str.split('.')
+            latitude_str_format[0] = commandmodule.create_fixed_length_packet_leading_padding(str(latitude_str_format[0]),
+                                                                     self.MAX_GPS_LATITUDE_LEADING_LEN, 0x30)
+            latitude_str_format[1] = commandmodule.create_fixed_length_packet_padding(str(latitude_str_format[1]),
+                                                                                              self.MAX_GPS_LATITUDE_TRAILING_LEN,
+                                                                                              0x30)
+            latitude_str = latitude_str_format[0] + '.' + latitude_str_format[1]
+
+        # Check Longitude formatting
         lon_check = len(longitude_str) <= self.MAX_GPS_LONGITUDE_LEN
         lon_dir_check = len(longitude_dir_str) <= self.MAX_GPS_LONGITUDE_DIR_LEN
```

Comment:
Again you never check for valid input values. They can only be "E" or "W".

![image](https://cloud.githubusercontent.com/assets/331540/23091225/dda5afac-f565-11e6-8a84-fd994e970576.png)